File: java/014_longest_common_prefix.java
// https://leetcode.com/problems/longest-common-prefix/

// Write a function to find the longest common prefix string amongst an array of strings.

public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0){
            return "";
        }
        int shortest_idx = 0;
        int shortest_len = Integer.MAX_VALUE;
        for (int i = 0; i < strs.length; i++){
            if (strs[i].length() < shortest_len){
                shortest_len = strs[i].length();
                shortest_idx = i;
            }
        }
        String prefix = "";
        for (int i = 0; i < shortest_len; i++){
            String s = strs[shortest_idx];
            char c = s.charAt(i);
            boolean flag = true;
            for (int j = 0; j < strs.length; j++){
                if (strs[j].charAt(i) != c){
                    flag = false;
                    break;
                }
            }
            if (flag){
                prefix = prefix + c;
            }
            else{
                break;
            }
        }
        return prefix;
    }
}

File: python/014_longest_common_prefix.py
# https://leetcode.com/problems/longest-common-prefix/

# Write a function to find the longest common prefix string amongst an array of strings.

class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        import sys
        shortest_idx = 0
        shortest_len = sys.maxint
        for i in range(0, len(strs)):
            if len(strs[i]) < shortest_len:
                shortest_len = len(strs[i])
                shortest_idx = i
        prefix = ""
        for i in range(0, shortest_len):
            char = strs[shortest_idx][i]
            flag = True
            for j in range(0, len(strs)):
                if strs[j][i] != char:
                    flag = False
                    break
            if flag:
                prefix = prefix+char
            else:
                break
        return prefix

x = Solution()
print x.longestCommonPrefix(["aaaaaa", "aabaab", "acaaac", "abcd", "ad"])